### PR TITLE
[Unified Checkpoint] Fix tie_weights save and load

### PR DESCRIPTION
**Pull request by DesmonDay**
### PR types
Bug fixes

### PR changes
Others

### Description
Fix tie_weights save and load.

### CODE REVIEW

**Review by JunnYu — DISMISSED** [review]
LGTM

File: paddlenlp/trainer/plugins/unified_checkpoint.py [hunk 1]
```diff
@@ -226,6 +226,10 @@ def load_unified_checkpoint_locally(args, model, resume_from_checkpoint: str, sa
 
     model_state_dict = get_expected_state_dict(model)
     expected_keys = set(list(model_state_dict.keys()))
+    if hasattr(model, "_tied_weights_keys") and model._tied_weights_keys is not None:
+        for key in model._tied_weights_keys:
+            expected_keys.remove(key)
```
Comment (ZHUI) [inline paddlenlp/trainer/plugins/unified_checkpoint.py]:
```suggestion 
    if model._keys_to_ignore_on_save is not None:
        for key in model._keys_to_ignore_on_save:
            expected_keys.remove(key)
```

Comment (ZHUI) [reply paddlenlp/trainer/plugins/unified_checkpoint.py]: https://github.com/PaddlePaddle/PaddleNLP/blob/1cd270a1556b1ab4895dc5d0b84e5fd57f20ef9e/paddlenlp/transformers/model_utils.py#L885-L886

我们的模型，应该有这个属性

Comment (ZHUI) [reply paddlenlp/trainer/plugins/unified_checkpoint.py]: 用 _keys_to_ignore_on_save 可能更好一点。

**Review by JunnYu — DISMISSED** [review]
LGTM

**Review by ZHUI — DISMISSED** [review]
LGTM

**Pull request merged**